helm_deploy/apply-for-legal-aid/templates/_envs.tpl

```diff
@@ -220,6 +220,21 @@ env:
       secretKeyRef:
         name: {{ template "apply-for-legal-aid.fullname" . }}
         key: sidekiqWebUiPassword
+  {{ if .Values.redis.enabled }}
+  - name: REDIS_PROTOCOL
+    value: "redis"
+  - name: REDIS_HOST
+  {{ if contains "redis" .Release.Name }}
```

Comment:
This is the only bit I find unclear... maybe some comments to expalin what the -redis-master difference is and why it's needed?

Reply:
so the [commit](https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/5825/commits/6f597d68010248502e80b5fd8ac7239699c50463)  explains somewhat but I was thinking to extract to a helper and can add a comment to that in any event. 

This took the most time time understand and resolve for 4505 work i would say. Essentially the redis chart contains this:

```yaml
{{/*
Create a default fully qualified app name.
We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
If release name contains chart name it will be used as a full name. <-- THIS IS THE ISSUE
*/}}
{{- define "common.names.fullname" -}}
{{- if .Values.fullnameOverride -}}
{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{- $name := default .Chart.Name .Values.nameOverride -}}
{{- if contains $name .Release.Name -}}
{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
{{- end -}}
{{- end -}}
{{- end -}}
```

Worth noting that this may well effect postgresql chart and generated host env var too, so that if you name a branch postgresql branch my-postgresql-branch it could cause issues. but not sure.